File: csharp/robot-name/RobotName.cs
using System;
using System.Collections.Generic;
using System.Text;

public class Robot
{
    private static HashSet<string> mhasNames = new HashSet<string>();

    private Random mobjRandom = null;
    private string mstrCurrentName = "";

    public Robot()
        {
            mobjRandom = new Random();
            Name = mGetNextName();
        }
    public string Name
    {
        get
        {
            return mstrCurrentName;
        }
        private set
        {
            mstrCurrentName = value;
        }
    }

    private string mGetNextName()
    {

    string strName = mGenerateName();
        while (mhasNames.Add(strName) == false)
        {
            strName = mGenerateName();
        }
        return strName;
    }
    private string mGenerateName()
    {
        StringBuilder objStringBuilder = new StringBuilder();

        objStringBuilder.Append((char)mobjRandom.Next((int)'A', (int)'Z' + 1));
        objStringBuilder.Append((char)mobjRandom.Next((int)'A', (int)'Z' + 1));
        objStringBuilder.Append(mobjRandom.Next(0, 10).ToString());
        objStringBuilder.Append(mobjRandom.Next(0, 10).ToString());
        objStringBuilder.Append(mobjRandom.Next(0, 10).ToString());
        return objStringBuilder.ToString();
    }

    public void Reset()
    {
        if(mhasNames.Contains(Name))
        {
            mhasNames.Remove(Name);
            Name = mGetNextName();
        }
    }
}
//exercism submit C:\Users\eric\Exercism\csharp\robot-name\RobotName.cs

File: python/robot-name/robot_name.py
import random
from random import randint
from string import ascii_uppercase

colNames = set()


class Robot:

    def __init__(self):
        self.name = self.generate_name()

    def generate_name(self) -> str:

        name = self.get_robot_name()

        while name in colNames:
            name = self.get_robot_name()

        colNames.add(name)
        return name

    def reset(self):
        self.name = self.generate_name()

    def get_robot_name(self) -> str:
        return "".join([
            random.choice(ascii_uppercase),
            random.choice(ascii_uppercase),
            str(randint(100, 999))
        ])

#submit C:\Users\eric\Exercism\python\robot-name\robot_name.py
